File: beispiele-einzeln/vier-seule/kapselung.java
class BankAccount {
    // private Eigenschaft
    private double balance;

    public BankAccount(double balance) {
        this.balance = balance;
    }

    // public Getter-Methode
    public double getBalance() {
        return balance;
    }

    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
        }
    }
}


// Die Verkapselung schützt die Daten der Klasse. 
// In diesem Beispiel ist das Feld balance privat und wird 
// über die Methoden getBalance und deposit aufgerufen.

File: beispiele-einzeln/vier-seule/kapselung.py
class BankAccount:
    def __init__(self, balance):
        self.__balance = balance  # private Eigenschaft

    # public Getter-Methode
    def get_balance(self):
        return self.__balance

    def deposit(self, amount):
        if amount > 0:
            self.__balance += amount


# Die Verkapselung schützt die Daten der Klasse. 
# In diesem Beispiel ist das Feld balance privat und wird 
# über die Methoden get_balance und deposit aufgerufen.